### DISCUSSION TITLE
TypeScript with a conditional navigator (Switch)

### DISCUSSION THREAD

**Discussion by sospedra** [react-navigation/react-navigation#10807] [Q&A]
It looks like these 2 docs cannot be combined:
* [Switch navigator](https://reactnavigation.org/docs/auth-flow#how-it-will-work)
* [TypeScript organization](https://reactnavigation.org/docs/typescript/#specifying-default-types-for-usenavigation-link-ref-etc)

I've been thinking about this and I can't come with a way to have a TS that can reflect this:

```tax
isSignedIn ? (
  <>
    <Stack.Screen name="Home" component={HomeScreen} />
    <Stack.Screen name="Profile" component={ProfileScreen} />
    <Stack.Screen name="Settings" component={SettingsScreen} />
  </>
) : (
  <>
    <Stack.Screen name="SignIn" component={SignInScreen} />
    <Stack.Screen name="SignUp" component={SignUpScreen} />
  </>
);
```

Am I missing something?

**Comment by chrisbianca** [comment]
@sospedra did you ever find a solution for this? I'm struggling to figure out a way to make this workable at the moment
